docs/cookbook/listening_to_core_events.md

```diff
@@ -90,14 +90,14 @@ This very simple example will just log the received signal.
 
 In `services.yml` (in your bundle):
 
-``` yaml
-parameters:
-    my.signal_listener.class: Acme\ExampleBundle\EventListener\SignalListener
- 
+``` yaml 
 services:
-    my.signal_listener:
-        class: '%my.signal_listener.class%'
-        arguments: ['@logger']
+    _defaults:
+        autowire: true
+        autoconfigure: false
```

Comment:
Nope :) See [Auto-tagging with autoconfigure doc](https://symfony.com/doc/current/service_container/3.3-di-changes.html#auto-tagging-with-autoconfigure). 

This means that we should either expand this section to include PHP code example or reference that to Symfony doc. The former one is preferred, because links to external services tend to stop working at some point ;)

Reply:
means the signal type should be checked in the Slot `recieve()` method. IMHO, I found the actual service configuration is more comfortable.

Reply:
@arfaram you're missing the point of defaulting `autoconfigure` to `true`. This is new best practice and should be encouraged even if some of our services don't support it yet.
Here by setting `autoconfigure: true` in `_defaults` you could skip
```yml
       tags:
            - { name: kernel.event_subscriber }
```
which is great benefit once you end up with a lot of such services.

> means the signal type should be checked in the Slot recieve() method. 

this is totally not related to autoconfiguration, see the comment that will follow shortly.